src/storage/src/hummock/local_version.rs

```diff
@@ -523,13 +529,24 @@ pub struct PinnedVersion {
 }
 
 impl PinnedVersion {
-    fn new(version: HummockVersion, unpin_worker_tx: UnboundedSender<HummockVersionId>) -> Self {
+    fn new(
+        version: HummockVersion,
+        pinned_version_manager_tx: UnboundedSender<PinVersionAction>,
+    ) -> Self {
         let version_id = version.id;
+
+        if pinned_version_manager_tx
+            .send(PinVersionAction::Pin(version.id))
+            .is_err()
+        {
+            tracing::warn!("failed to send unpin {}", version_id);
+        }
+
         PinnedVersion {
             version: Arc::new(version),
             guard: Arc::new(PinnedVersionGuard {
```

Comment:
The `new_pin_version` below will also instantiate a `PinnedVersionGuard` and we also need to send `Pin` there.

I think we may introduce a `new` method for `PinnedVersionGuard` and send the `Pin` in it, and remove all `PinnedVersionGuard` instantiation via struct.